### ISSUE TITLE
ERROR: ErrorResult lib path is required

### ISSUE THREAD

**Issue by SandPod** [kawa1214/import-lint#40]
### Observed behavior

When running `dart analyze --fatal-infos` in GitHub actions in a closed source repository, `import_lint` as a analyzer plugin, the analyzer crashes with the following stack trace:

```bash
ERROR: ERROR: An unexpected error was encountered by the Analysis Server.
Please file an issue at https://github.com/dart-lang/sdk/issues/new/choose with the following details:

ErrorResult lib path is required
#0      new FilePathResourceLocator.fromUri (package:import_lint/src/analyzer/resource_locator.dart:75:7)
#1      DriverBasedAnalysisContextAnalyzer.analyzeFile (package:import_lint/src/analyzer/analyzer.dart:38:61)
<asynchronous suspension>
#2      ImportLintPlugin.analyzeFile (package:import_lint/src/plugin/plugin.dart:36:22)
<asynchronous suspension>
#3      ServerPlugin.analyzeFiles (package:analyzer_plugin/plugin/plugin.dart:119:7)
<asynchronous suspension>
#4      ServerPlugin.afterNewContextCollection.<anonymous closure> (package:analyzer_plugin/plugin/plugin.dart:86:7)
<asynchronous suspension>
#5      ServerPlugin._forAnalysisContexts (package:analyzer_plugin/plugin/plugin.dart:525:7)
<asynchronous suspension>
#6      ServerPlugin.afterNewContextCollection (package:analyzer_plugin/plugin/plugin.dart:84:5)
<asynchronous suspension>
#7      ServerPlugin.handleAnalysisSetContextRoots (package:analyzer_plugin/plugin/plugin.dart:273:5)
<asynchronous suspension>
#8      ServerPlugin._getResponse (package:analyzer_plugin/plugin/plugin.dart:544:18)
<asynchronous suspension>
#9      ServerPlugin._onRequest (package:analyzer_plugin/plugin/plugin.dart:611:18)
<asynchronous suspension>
```

### Expected behavior
The analyzer should execute correctly.

### Preliminary debugging
This was first reported to the dart sdk here: https://github.com/dart-lang/sdk/issues/59744

It was suggested that the check if we are inside the `lib` directory might be incorrect: https://github.com/dart-lang/sdk/issues/59744#issuecomment-2567864889

### Additional context

Plugin is defined like so:
```yaml
analyzer:
  plugins:
    - import_lint
```

And multiple `no_import_feature_from_domain` rules are specified: 
```yaml
import_lint:
  severity: "warning"
  rules:
    no_import_feature_from_domain:
      target: ....
      from: ....
      except: []
```

- This started happening when we upgraded to Dart `3.6.0`.
- This is used in a fairly large project.
- This does not happen when running the same command locally.

### Project info
- Dart version: `Dart SDK version: 3.6.0 (stable) (Thu Dec 5 07:46:24 2024 -0800) on "linux_x64"`
- Runs on: `Ubuntu 24.04.1 LTS`

**Comment by christerswahn** [comment]
Hi @kawa1214! This is a useful tool for us but this issue stops us from using it in CI. Any progress on this?
